### PR TITLE
Support for NUCLEO_F401RE + X-NUCLEO_IDW01M1 WiFi

### PR DESCRIPTION
**Pull request by JanneKiiskila**
Combination of commits by betwz and some changes on top of that by me to get the NUCLEO F401RE + X-Nucleo WiFi working.

### CODE REVIEW

File: easy-connect.lib [hunk 1]
```diff
@@ -1 +1 @@
-https://github.com/ARMmbed/easy-connect/#bf821b0695ccada3620a6091503eb53a2a1ac3c8
+https://github.com/ARMmbed/easy-connect/#adb235d929815d92083aa2abeed9f29b0386b2c3
```
Comment (betzw) [inline easy-connect.lib]: Update with latest `easy-connect` hash code (once merged in).

Comment (JanneKiiskila) [reply easy-connect.lib]: Yep!

**Comment by JanneKiiskila** [comment]
Latest easy-connect has been updated.

**Comment by JanneKiiskila** [comment]
@betzw  - there is now some problem with the latest one, it doesnt' connect at all. With the previous easy-connect it was still able to connect - I think I need to have something more now as you have multiple models, but how do you know what? I think the README.md has to be updated a bit to detail this out.

**Comment by betzw** [comment]
@JanneKiiskila - could you pls. elaborate on this a bit more?

**Comment by JanneKiiskila** [comment]
With the latest x-nucleo-wifi-idw01m1 repo -> it fails the connection. I've double checked by wifi APN + passphrase and they are OK, but it fails. 

```
Starting mbed Client example
[EasyConnect] IPv4 mode
[EasyConnect] Using WiFi (X-NUCLEO-IDW01M1)
[EasyConnect] Connecting to WiFi jannek.iki.fi
[EasyConnect] MAC address 00:80:E1:B7:D0:AC
[EasyConnect] Connection to Network Failed -3012!

Connection to Network Failed - exiting application...
```

The text on my Wifi module is stating X-NUCLEO-IDW01M1, so the default value in the `mbed_lib.json` should match the HW.

**Comment by betzw** [comment]
@JanneKiiskila, you might want to compare with PR #324 and picky-pick necesary modifications.

**Comment by betzw** [comment]
Any progress on this or PR #327?

cc @screamerbg, @nikapov-ST

**Comment by betzw** [comment]
Bump @JanneKiiskila

File: configs/eth-wifi-mbedignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
 easy-connect/atmel-rf-driver/*
 easy-connect/mcr20a-rf-driver/*
 easy-connect/stm-spirit1-rf-driver/*
+easy-connect/wifi-x-nucleo-idw01m1/*
```
Comment (JammuKekkonen) [inline configs/eth-wifi-mbedignore]: at least esp8266 / k64f seems to compile fine without this. is there some known combination that needs this?

File: configs/wifi-esp8266-mbedignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+easy-connect/atmel-rf-driver/*
```
Comment (JammuKekkonen) [inline configs/wifi-esp8266-mbedignore]: at least esp8266 / k64f seems to compile fine without this. is there some known combination that needs this?

File: configs/wifi-idw01m1-mbedignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+easy-connect/atmel-rf-driver/*
```
Comment (JammuKekkonen) [inline configs/wifi-idw01m1-mbedignore]: at least esp8266 / k64f seems to compile fine without this. is there some known combination that needs this?

**Comment by JammuKekkonen** [comment]
@JanneKiiskila functionality works for me, but i'd remove the (afaik) unnecessary mbedignores unless there are some known cases where they are needed.

**Comment by betzw** [comment]
In general I would consider this PR as a bit stale and prefer PR #327 to be considered for merging (at the place of this PR). @JanneKiiskila, what do you think?

Regarding the `mbedignore` files, I do not know what ARM's policy is there, but I would prefer to exclude as much unnecessary code as possible from builds ...

**Comment by JanneKiiskila** [comment]
Agree with @betzw - turns out that for example ATParser leaves 35 bytes of code there, if it's compiled and not used.

**Comment by teetak01** [comment]
Closing in favor of: https://github.com/ARMmbed/mbed-os-example-client/pull/327

**Pull request closed without merging**